### PR TITLE
vscode-extensions.dbaeumer.vscode-eslint: 2.2.6 -> 2.4.0

### PR DESCRIPTION
**Pull request by robbevp**
###### Description of changes

Bump Eslint Vscode extension from 2.2.6 to 2.4.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by datafoo — COMMENTED** [review]
As of today, all versions after 2.2.6 are pre-releases (https://marketplace.visualstudio.com/items/dbaeumer.vscode-eslint/changelog).

I am not sure we should replace a stable release with a pre-release.

**Comment by robbevp** [comment]
> As of today, all versions after 2.2.6 are pre-releases (https://marketplace.visualstudio.com/items/dbaeumer.vscode-eslint/changelog).
> 
> I am not sure we should replace a stable release with a pre-release.

Ah, I had understood the 2.4.0 version to be a stable release (with everything in 2.3.x being pre-release).

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -753,8 +753,8 @@ let
         mktplcRef = {
           name = "vscode-eslint";
           publisher = "dbaeumer";
-          version = "2.2.6";
-          sha256 = "sha256-1yZeyLrXuubhKzobWcd00F/CdU824uJDTkB6qlHkJlQ=";
+          version = "2.4.0";
+          sha256 = "02q5g5p2vy7yxm2l89zvlkfx4rkdyzrrjaa97gl5sf6g88k11igc";
```
Comment (datafoo) [inline pkgs/applications/editors/vscode/extensions/default.nix:757]: How did you generate the hash? they should start by "sha256-" now.

Comment (robbevp) [reply pkgs/applications/editors/vscode/extensions/default.nix:757]: I used `./path-to-nixpkgs/pkgs/applications/editors/vscode/extensions/update_installed_exts.sh ` This gave me the sha i used.

I can build the package both with that output and the sha you provided

Comment (datafoo) [reply pkgs/applications/editors/vscode/extensions/default.nix:757]: Yes, I think "SRI hashes" is preferred according to https://nixos.wiki/wiki/Nix_Hash#Usage

Comment (robbevp) [reply pkgs/applications/editors/vscode/extensions/default.nix:757]: Updated to use the SRI hash

**Review by datafoo — APPROVED** [review]

**Review by superherointj — APPROVED** [review]

**Comment by datafoo** [comment]
Thank you for the PR @robbevp 

If it is backward compatible with version 2.2.6 then you could backport the change on the channel 22.11 by simply adding the label `backport release-22.11` on this PR.

**Comment by robbevp** [comment]
Hi @datafoo

I'm not able to add additional labels to this PR. The changes here are definitely backward compatible, so if you can add the label, please do!

**Pull request merged**